just_curious_game.ex

```diff
@@ -0,0 +1,325 @@
+                ...
+
+    :submissions # sorted set
+                :question-o-matic-1 -> 2 # a question-o-matic submitter with score
+                :U111               -> 1 # or real user submitter
+                :U222               -> 0
+
+    :nominatables # hash
+                 :000.001 -> U111 # nominatable message id and user id
+                 :000.002 -> U222
+    :nominations
+                :upvotes
+                          :U111
+                               :target -> "000.002" # user 1 upvoted second nominatable
+                               :reaction -> "heart" # with a heart reaction
+                          :U222
+                               :target -> "000.001"
+                               :reaction -> "+1" # with a heart reaction
+                :downvotes
+                          :U111 -> "000.003"
+
+  ```
+  """
+
+  @spec create(String.t, Map.t) :: %JustCurious.Game{}
+  def create(team, opts \\ %{}) do
+    name       = generate_name(team, :game)
+    game_id    = name |> String.downcase |> String.replace(" ", "-")
+    created_at = DateTime.utc_now |> DateTime.to_unix(:microseconds)
+
+    new_game = %JustCurious.Game{
+      team: team,
+      name: name,
+      channel: game_id,
+      round: opts[:round],
+      state: opts[:state] || :new,
+      created_at: created_at,
+      chainlink: 0
+    } |> Map.merge(JustCurious.GameConfigs.get(opts[:config]))
+
+    incr("team:#{team}:games_played")
+    hmset("team:#{team}:game:#{game_id}", redis_list_from_struct(new_game))
+
+    JustCurious.MessageQueue.deliver_at(new_game, "transition start", created_at + new_game.start_timeout)
+
+    sadd("team:#{team}:games", game_id)
+    (opts[:players] || []) |> Enum.each(&(add_participant(new_game, &1)))
+    (opts[:joins] || []) |> Enum.each(&(add_participant(new_game, &1, :join)))
+    (opts[:invites] || []) |> Enum.each(&(add_participant(new_game, &1, :invite)))
+    (opts[:spectators] || []) |> Enum.each(&(add_participant(new_game, &1, :spectator)))
+
+    hincrby("team:#{team}:game:#{game_id}", "chainlink", 0)
+
+    find(team, game_id)
+  end
+
+  @spec find(String.t, String.t) :: %JustCurious.Game{}
+  def find(team_id, game_id) do
+    # TODO game does not exist
+    game = struct(JustCurious.Game, string_map_to_atom_map(hgetall("team:#{team_id}:game:#{String.downcase("#{game_id}")}")))
+    game
+      |> Map.put(:participants, participants(game))
+      |> Map.put(:players, players(game))
+      |> Map.put(:spectators, spectators(game))
+      |> Map.put(:invites, invites(game))
+      |> Map.put(:joins, joins(game))
+  end
+
+  ## STATE/ROUND/CHAINLINK ##
+  @spec get_state(%JustCurious.Game{}) :: atom
+  def get_state(game) do
+    hget("team:#{game.team}:game:#{game}", "state") |> String.to_atom
+  end
+
+  @spec set_state(%JustCurious.Game{}, atom) :: atom
+  def set_state(game, state) do
+    hset("team:#{game.team}:game:#{game}", "state", state)
+    :"#{state}"
```

Comment:
If your `@spec` is correct, why not directly returning `state` without string interpolation?
If it breaks, it means that the `atom` part of the `@spec` might be wrong.